packages/core/src/RenderingEngine/vtkClasses/vtkOffscreenMultiRenderWindow.js

```diff
@@ -39,6 +43,13 @@ function vtkOffscreenMultiRenderWindow(publicAPI, model) {
   model.openGLRenderWindow = vtkStreamingOpenGLRenderWindow.newInstance();
   model.renderWindow.addView(model.openGLRenderWindow);
 
+  // Create offscreen canvas and set it immediately
+  const offscreenCanvas = document.createElement('canvas');
+  offscreenCanvas.width = MAX_WIDTH;
```

Comment:
as discussed, let's detect the device isMobile and cap it at 4096 by 4096 based on this link https://jhildenbiddle.github.io/canvas-size/#/?id=test-results